File: edit/edit.py
def levenshtein_distance(X, Y):
    N, M = len(X), len(Y)
    dp = [[0 for _ in range(M+1)] for _ in range(N+1)]

    for i in range(N+1):
        dp[i][0] = i

    for j in range(M+1):
        dp[0][j] = j

    for i in range(1, N+1):
        for j in range(1, M+1):
            if X[i-1] == Y[j-1]:
                dp[i][j] = dp[i-1][j-1]
            else:
                dp[i][j] = min(dp[i-1][j], dp[i][j-1], dp[i-1][j-1]) + 1

    return dp[N][M]

# Lettura dell'input
with open('input.txt', 'r') as file:
    input_data = file.readlines()

# Elaborazione dei task e scrittura dell'output
output_data = []
for line in input_data:
    N, M, X, Y = line.split()
    N, M = int(N), int(M)
    distance = levenshtein_distance(X, Y)
    output_data.append(str(distance))

with open('output.txt', 'w') as file:
    file.write('\n'.join(output_data))

File: edit/edit.cpp
#include <fstream>
#include <iostream>
#include <vector>

#define TASK 1

using namespace std;

int edit(int sl1, int sl2, string s1, string s2) {
  vector<vector<int>> matrix(sl1 + 1, vector<int>(sl2 + 1, 0));

  for (int i = 0; i < sl1 + 1; i++)
    matrix[i][0] = i;
  for (int j = 0; j < sl2 + 1; j++)
    matrix[0][j] = j;

  for (int i = 1; i < sl1 + 1; i++)
    for (int j = 1; j < sl2 + 1; j++)
      if (s1[i - 1] == s2[j - 1])
        matrix[i][j] = matrix[i - 1][j - 1]; // match
      else
        matrix[i][j] =
            min(matrix[i - 1][j - 1], min(matrix[i - 1][j], matrix[i][j - 1])) +
            1;
for(int i = 0;i<sl1+1;i++){
	for(int j=0;j<sl2+1;j++)
		cout << matrix[i][j] << " ";
	cout << endl;
}
  return matrix[sl1][sl2];
}
int main() {
  ifstream input("input.txt");
  ofstream output("output.txt");

  int sl1;
  int sl2;
  string s1;
  string s2;

  for (int i = 0; i < TASK; i++) {
    input >> sl1 >> sl2 >> s1 >> s2;
    output << edit(sl1, sl2, s1, s2) << endl;
  }
}
